### DISCUSSION TITLE
Auto subscribing events

### DISCUSSION THREAD

**Discussion by le-kote** [space-wizards/space-station-14#33430] [Technical]
It would be great if there was a way to automatically subscribe events if they had only their abstract parent subscribed. 
For example, inventory relayed events: now you have to manually subscribe them because you can not make them subscribed because of IInventoryRelayedEvent parent. It would be much easier if you could just register IInventoryRelayedEvent instead of copypasting registration for every single event

**Comment by deltanedas** [comment]
for relayed events thats for drop-in compatibility with regular event handlers, you are probably supposed to just write a system to handle the RelayedEvent directly
